### DISCUSSION TITLE
Array type with windowing like functionality?

### DISCUSSION THREAD

**Discussion by thadguidry** [ohler55/ojg#144] [Ideas]
Not sure if windowing is the right analogy.  I was reading through notes on a recent merge in DuckDB and saw potentially some useful ideas that might be added to OjG to optimize on nested data through Arrays.  I am not sure of what or all of the capabilities listed in the merge therein but wanted to use this issue to ask the question of "any useful ideas here that OjG can incorporate or in the future?"

Helping OjG becoming even faster with nested data.

https://github.com/duckdb/duckdb/pull/8983

**Comment by ohler55** [comment]
This is probably more of a discussion item but I'll take a look.

**Comment by thadguidry** [comment]
You can convert it to a discussion indeed.  GitHub has a way to do that now I think.

**Comment by ohler55** [comment]
This is now a discussion thread. 

After reading the duckdb PR I'm not sure how it could be applied to OjG. Can you give a few specifics about what caught your eye?

**Reply by thadguidry** [reply]
Yes, I was thinking about making child element access faster specifically through filtering.  And we now have 'in' operator (recall our gmail private thread)

**Reply by ohler55** [reply]
I don't think the use of fixed size arrays and filtering fits in the current code. I might still be missing what you8 had in mind but so far I can't see a use.

**Reply by thadguidry** [reply]
No worries.  You might be right.  This discussion might be useful later, so at least we have it for posterity.
